### PR TITLE
tracking clicks on link to start guide

### PR DESCRIPTION
**Pull request by 2is10**
Also tracking which page each event occurs on. This used to be done on our pre-angular site, but was lost when we launched the original angular site.

We can’t do this automatically without modifying the angulartics source code, so I’m manually adding a `path` property for each event, then translating that to a `type` (page kind) when the event reaches our mixpanel code.

@martinraison @stkem

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular-black/src/layout/rightCol/RightColCtrl.js [hunk 1]
```diff
@@ -31,6 +31,7 @@ angular.module('kifi.layout.rightCol', ['kifi.modal'])
     $scope.yesLikeHelpRank = function () {
       $scope.data.showHelpRankHelp = false;
       $analytics.eventTrack('user_clicked_page', {
+        'type': 'HelpRankHelp',
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular-black/src/layout/rightCol/RightColCtrl.js:34]: Note: Sometimes we set `type` directly, like when a link is inside a dialog. In this case, the `type` should still match the `type` used in the corresponding `user_viewed_page` event.

cc @raymondng

File: kifi-backend/modules/shoebox/angular-black/src/layout/rightCol/RightColCtrl.js [hunk 2]
```diff
@@ -38,27 +39,22 @@ angular.module('kifi.layout.rightCol', ['kifi.modal'])
...
     };
 
-    $scope.showClicks = function() {
-      $location.path('/helprank/click');
+    $scope.trackClickOnClicks = function() {
       $analytics.eventTrack('user_clicked_page', {
-        'action': 'helpRankClicks'
-      });
-      $analytics.eventTrack('user_viewed_page', {
-        'action': 'helpRankClicks'
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular-black/src/layout/rightCol/RightColCtrl.js:51]: Note that here I removed the `user_viewed_page` tracking and the `$location.path('/helprank/click')` call. For links that modify the URL in the location bar, it’s better to put the link path directly in the link’s `href` attribute. Angular correctly (quickly) handles these using the history API, page view tracking will occur automatically, and using the actual `href` gives the user a better indication about what will happen when the link is clicked.

File: kifi-backend/modules/shoebox/angular-black/src/tags/tagService.js [hunk 3]
```diff
@@ -56,6 +53,11 @@ angular.module('kifi.tagService', [
         }
         return res;
       });
+      $analytics.eventTrack('user_clicked_page', {
+        'action': 'addKeepsToTag',
+        'path': $location.path()
+      });
+      return post;
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular-black/src/tags/tagService.js:60]: Notice that I moved a bunch of event tracking out of HTTP response handlers, and instead put them immediately after the action HTTP request is initiated. The intent of our analytics is to track real actions (e.g. clicks), not just actions that we handle successfully.

File: kifi-backend/modules/shoebox/angular-black/src/common/services/mixpanelService.js [hunk 4]
```diff
@@ -94,20 +96,22 @@
       $analyticsProvider.registerPageTrack(function (path) {
         if (profileService && $window) {
           var mixpanel = $window.mixpanel;
-          var normalizedPath = getLocation(path);
```
Comment (2is10) [inline kifi-backend/modules/shoebox/angular-black/src/common/services/mixpanelService.js:97]: We were attempting path translation twice here (before the `pageTrack...` calls and inside them), so I removed this one.

**Comment by martinraison** [comment]
Cool! I didn't realize we had direct page view tracking (as opposed to replacing `href` with custom javascript). Thanks - looks good!

File: kifi-backend/modules/shoebox/angular-black/src/layout/rightCol/RightColCtrl.js (hunk 2 above)
Comment (raymondng) [inline kifi-backend/modules/shoebox/angular-black/src/layout/rightCol/RightColCtrl.js:51]: Actually that's what the original code does, and yes it works well.

However, recent product requirement (per @JRuff42 ) is that we send both (viewed and clicked) events, hence the change. AFAIK using $location is fully supported -- but yes I agree that using `href` has better clarity. In short, as long as product is happy, I am good.

Comment (raymondng) [inline kifi-backend/modules/shoebox/angular-black/src/layout/rightCol/RightColCtrl.js:51]: Oops I missed the part about auto page view tracking. That's great! :+1:

**Pull request merged**